regression: parameter type may not live long enough

See https://crater-reports.s3.amazonaws.com/beta-1.62-1/beta-2022-05-20/gh/FSMaxB.rust-genealogy/log.txt:

```
[INFO] [stdout] error[E0311]: the parameter type `Extracted` may not live long enough
[INFO] [stdout]   --> genealogy-java-apis/src/comparator.rs:24:13
[INFO] [stdout]    |
[INFO] [stdout] 24 |             compare: Box::new(move |left, right| match (self.compare)(left, right) {
[INFO] [stdout]    |                      ^^^^^^^^ ...so that the type `Extracted` will meet its required lifetime bounds...
[INFO] [stdout]    |
[INFO] [stdout] note: ...that is required by this bound
[INFO] [stdout]   --> genealogy-java-apis/src/comparator.rs:21:28
[INFO] [stdout]    |
[INFO] [stdout] 21 |         for<'a> Extracted: Ord + 'a,
[INFO] [stdout]    |                                  ^^
[INFO] [stdout] help: consider adding an explicit lifetime bound...
[INFO] [stdout]    |
[INFO] [stdout] 21 |         for<'a> Extracted: Ord + 'a + 'a,
[INFO] [stdout]    |                                     ++++
[INFO] [stdout] 
[INFO] [stdout] 
[INFO] [stdout] error: aborting due to previous error
```

This is a 1.62 regression, it seems, based on Crater. At the very least, it's a diagnostics issue that we're suggesting adding an already existing lifetime bound.

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_DATA_START$${"user":"Rageking8"}$$TRIAGEBOT_ASSIGN_DATA_END -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/src/test/ui/closures/issue-97607.rs
+++ b/src/test/ui/closures/issue-97607.rs
@@ -1,0 +1,12 @@
+// check-pass
+#[allow(unused)]
+
+fn test<T, F, U>(f: F) -> Box<dyn Fn(T) -> U + 'static>
+where
+    F: 'static + Fn(T) -> U,
+    for<'a> U: 'a, // < This is the problematic line, see #97607
+{
+    Box::new(move |t| f(t))
+}
+
+fn main() {}

